src/components/RegisterForm/RegisterForm.js

```diff
@@ -26,103 +43,225 @@ class RegisterForm extends Component {
     });
   }
 
-  saveToLocalStore() {
+  setFirstName(val) {
+    this.setState({ firstName: val });
+  }
 
+  setLastName(val) {
+    this.setState({ lastName: val });
   }
 
-  handleSubmit(e) {
-    e.preventDefault();
-    const form = e.target;
-    const data = new FormData(form);
+  setEmail(val) {
+    this.setState({ email: val });
+  }
 
-    const URL = '/api/form-submit-url';
+  setPassword(val) {
+    this.setState({ password: val });
+  }
 
-    fetch(URL, {
-      method: 'POST',
-      body: data
-    });
+  setCompany(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, company: val } });
+  }
+
+  setAddress(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, addressOne: val } });
+  }
+
+  setCity(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, city: val } });
+  }
+
+  setPostalCode(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, postalCode: +val } });
+  }
+
+  setPhone(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, phone: +val } });
+  }
+
+  selectCountry(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, country: val } });
+  }
+
+  selectRegion(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, region: val } });
+  }
+
+  validate() {
+    const { showMessage } = this.props;
+
+    let isInputValid = false;
+
+    isInputValid = this.formRegister.current.reportValidity();
+
+    if (!isInputValid) {
+      showMessage(VALIDATION_FAILED);
+    } else {
+      this.sendData();
+    }
+
+    return isInputValid;
   }
 
+
+  async sendData() {
+    const { showMessage } = this.props;
+    const userAPI = new HttpService();
+    const signupURL = `${process.env.SERVER_URL}/signup`;
+    const {
+      firstName,
+      lastName,
+      email,
+      password,
+      address
+    } = this.state;
+    const userData = {
+      firstName,
+      lastName,
+      email,
+      password,
+      address
+    };
+
+    try {
+      const response = await userAPI.post(signupURL, userData);
+      const errorMessage = response.data.message;
+
+      if (errorMessage) {
+        showMessage(REGISTER_FAILED);
+      } else {
+        showMessage(REGISTER_SUCCESS);
+      }
+    } catch (error) {
+      showMessage(REGISTER_FAILED);
+    }
+  }
+
+
   render() {
-    const { email } = this.state;
+    const { email, address: { country, region } } = this.state;
 
     return (
-      <form className={`${CN}`} onSubmit={this.handleSubmit}>
+      <form
+        className={cx(CN)}
```

Comment:
You can write just this: className={CN}, because you are not using any functionality from the Classnames library here (and you can remove the import of this library in the top of this file then).